src/core/lib/transport/metadata.h

```diff
@@ -132,17 +134,114 @@ void* grpc_mdelem_get_user_data(grpc_mdelem md, void (*if_destroy_func)(void*));
 void* grpc_mdelem_set_user_data(grpc_mdelem md, void (*destroy_func)(void*),
                                 void* user_data);
 
+void grpc_mdelem_trace_ref(void* md, const grpc_slice& key,
+                           const grpc_slice& value, gpr_atm* refcnt,
+                           const char* file, int line);
+void grpc_mdelem_trace_unref(void* md, const grpc_slice& key,
+                             const grpc_slice& value, gpr_atm* refcnt,
+                             const char* file, int line);
+
+typedef void (*destroy_user_data_func)(void* user_data);
+
+struct UserData {
+  gpr_mu mu_user_data;
+  gpr_atm destroy_user_data;
+  gpr_atm user_data;
+};
+
+typedef struct interned_metadata {
```

Comment:
this is C++, so you can drop the `typedef` from old C code and use `struct TypeName` here and below.

Reply:
Done.